kivy/logger.py

```diff
@@ -150,47 +151,40 @@ class FileHandler(logging.Handler):
     encoding = 'utf-8'
 
     def purge_logs(self):
-        '''Purge log is called randomly to prevent the log directory from being
-        filled by lots and lots of log files.
-        You've a chance of 1 in 20 that purge log will be fired.
-        '''
-        if randint(0, 20) != 0:
-            return
+        """Purge logs which exceed the maximum amount of log files,
+        starting with the oldest creation timestamp (or edit-timestamp on Linux)
+        """
+
         if not self.log_dir:
             return
 
         from kivy.config import Config
-        maxfiles = Config.getint('kivy', 'log_maxfiles')
-
-        if maxfiles < 0:
-            return
+        maxfiles = Config.getint("kivy", "log_maxfiles")
 
-        Logger.info('Logger: Purge log fired. Analysing...')
-        join = os.path.join
-        unlink = os.unlink
+        # Get path to log directory
+        log_dir = pathlib.Path(self.log_dir)
 
-        # search all log files
-        lst = [join(self.log_dir, x) for x in os.listdir(self.log_dir)]
-        if len(lst) > maxfiles:
-            # get creation time on every files
-            lst = [{'fn': x, 'ctime': os.path.getctime(x)} for x in lst]
+        if maxfiles < 0:  # No log file limit set
+            return
 
-            # sort by date
-            lst = sorted(lst, key=lambda x: x['ctime'])
+        Logger.info("Logger: Purge log fired. Processing...")
 
-            # get the oldest (keep last maxfiles)
-            lst = lst[:-maxfiles] if maxfiles else lst
-            Logger.info('Logger: Purge %d log files' % len(lst))
+        # Get all files from log directory and corresponding creation timestamps
+        files = [(item, item.stat().st_ctime) for item in log_dir.iterdir() if item.is_file()]
```

Comment:
```suggestion
        files = [(item, item.stat().st_ctime) 
                   for item in log_dir.iterdir() if item.is_file()]
```
Line was too long (>80).